macros/edr/tests/test_exposure_schema_validity.sql

```diff
@@ -0,0 +1,64 @@
+    {%- set exposures = ( exposures | default(graph.exposures) ).values() -%}
+    {%- set columns = columns | default(adapter.get_columns_in_relation(model))  -%}
+
+    {%- set model_relation = elementary.get_model_relation_for_test(model, context["model"]) -%}
+    {%- set full_table_name = elementary.relation_to_full_name(model_relation) -%}
+    {{- elementary.test_log('start', full_table_name, 'exposure validation') -}}
+
+    {%- set matching_exposures = [] -%}
+
+    {%- for exposure in exposures -%}
+        {%- if node in exposure.depends_on.nodes and exposure['meta'] | default(none) is not none -%}
+            {%- do matching_exposures.append(exposure) -%}
+        {%- endif -%}
+    {%- endfor -%}
+    {%- if matching_exposures | length > 0 -%}
+        {%- set columns_dict = {} -%}
+        {%- for column in columns -%}
+            {%- do columns_dict.update({ column['name'] | upper : elementary.normalize_data_type(column['dtype']) }) -%}
+        {%- endfor -%}
+        {%- set invalid_exposures = [] -%}
+        {%- for exposure in matching_exposures -%}
+            {# Depend on meta since column level info is not available on exposures #}
+            {%- set meta = exposure['meta'] | default(none) -%}
+            {%- if meta != none and meta['columns'] | default(none) is iterable -%}
+                {%- for column in meta['columns'] -%}
+                    {%- if matching_exposures | length == 1 or context['render'](column['source']) | default('') == node -%}
+                        {%- if column['name'] | upper not in columns_dict.keys() -%}
+                            {%- do invalid_exposures.append({
+                                    'exposure': exposure['name'],
+                                    'url': exposure['url'],
+                                    'error': column['name'] ~ ' column missing in the model'
+                                    })
+                            -%}
+                        {%- elif column['data_type'] | default('') != '' and column['data_type'] != columns_dict[column['name'] | upper] -%}
+                            {%- do invalid_exposures.append({
+                                    'exposure': exposure['name'],
+                                    'url': exposure['url'],
+                                    'error': 'different data type for the column ' ~ column['name'] ~ ' ' ~ column['data_type'] ~ ' vs ' ~ columns_dict[column['name'] | upper]
+                                    })
+                            -%}
+                        {%- endif -%}
+                    {%- endif -%}
+                {%- endfor -%}
+            {%- endif -%}
+        {%- endfor -%}
+        {%- if invalid_exposures | length > 0 -%}
+            {%- for invalid_exposure in invalid_exposures %}
+                {{ 'UNION ALL ' if not loop.first }}SELECT '{{ invalid_exposure['exposure'] }}' as exposure, '{{ invalid_exposure['url'] }}' as url, '{{ invalid_exposure['error'] }}' as error
```

Comment:
I think it's better to use [`exceptions.raise_compiler_error`](https://docs.getdbt.com/reference/dbt-jinja-functions/exceptions#raise_compiler_error) because query results are not presented while exceptions are.